solutions/gold/usaco-696.mdx

```diff
@@ -306,10 +307,117 @@ int main() {
 </CPPSection>
 </LanguageSection>
 
-<IncompleteSection>
+## Solution 3: HLD
 
-Add Centroid Decomp/HLD Solution
+**Time Complexity:** $\mathcal{O}(N\log ^2N)$
+
+Sort the nodes from largest to smallest value, do path updates from each node to the root.
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+class segTree {
```

Comment:
consider using the iterative sum segtree in the [purs module](https://usaco.guide/gold/PURS?lang=cpp), as I think that impl is cleaner and also probably more familiar